### ISSUE TITLE
Uncaught TypeError: Super expression must either be null or a function

### ISSUE THREAD

**Issue by devdsc** [NightCatSama/vue-slider-component#410]
#### Describe the bug
I am receiving below error on latest version ( to be specific 3.0.2 and above versions)

vue-slider-component.umd.min.js?4971:1820 Uncaught TypeError: Super expression must either be null or a function
    at d (vue-slider-component.umd.min.js?4971:1820)
    at eval (vue-slider-component.umd.min.js?4971:2065)
    at Module.fb15 (vue-slider-component.umd.min.js?4971:2068)
    at r (vue-slider-component.umd.min.js?4971:15)
    at eval (vue-slider-component.umd.min.js?4971:63)
    at eval (vue-slider-component.umd.min.js?4971:65)
    at eval (vue-slider-component.umd.min.js?4971:3)
    at eval (vue-slider-component.umd.min.js?4971:9)
    at Object../node_modules/vue-slider-component/dist/vue-slider-component.umd.min.js (manifest.build.js:3551)

#### Environment (If you feel unrelated, please delete the block)

 - OS & Version:Windows all browsers
 - Vue version: vue.runtime.min.js?v=v2.4.2
 - Component Version: v3.0.2 and above

**Comment by NightCatSama** [comment]
How did you use it?

**Comment by Digifoxman** [comment]
Hi there,

Nice work, slider looks really neat!

I am getting this error as well in my console:

![image](https://user-images.githubusercontent.com/6566578/70379764-010de080-1931-11ea-9ce8-1a6961051d86.png)

But the slider itself (I wrapped it in a simple Vuejs component) seems to work fine. No further errors or warnings.

My environment:

(index.hml)
    \<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/vue-slider-component@latest/theme/default.css">
    <script src="https://cdn.jsdelivr.net/npm/vue-slider-component@latest/dist/vue-slider-component.umd.min.js"></script>

(package.json)
    "vue": "^2.6.10",
    "vue-slider-component": "^3.0.45",

(OS)
   Ubuntu 16.04.3 LTS

Best regards, Ulfert.

**Comment by dlueth** [comment]
Same problem here, in an AMD-Environment @NightCatSama

**Comment by NightCatSama** [comment]
@dlueth  See if this issue is helpful to you.

<https://github.com/NightCatSama/vue-slider-component/issues/379>

**Comment by dlueth** [comment]
@NightCatSama was going to try that out tonight and report back. But my guess would be: it should. But that would mean that your umd-part is broken for amd

**Comment by dlueth** [comment]
@NightCatSama Yes, as expected, it works perfectly fine when I add "vue" as a dependency manually to the UMD block like this:

`if(typeof exports === 'object' && typeof module === 'object')
        module.exports = factory(require("vue"));
    else if(typeof define === 'function' && define.amd)
        define(["vue"], factory);
    else if(typeof exports === 'object')
        exports["vue-slider-component"] = factory(require("vue"));
    else
        root["vue-slider-component"] = factory(root["Vue"]);`

**Comment by NightCatSama** [comment]
@dlueth Thanks for your report, this bug is fixed in v3.1.5.

**Comment by dlueth** [comment]
Thanks!

**Comment by ShirinShirinak** [comment]
This issue still exists

**Comment by wizzer129** [comment]
Issue exists for me in 3.2.15

**Comment by miayang0513** [comment]
@wizzer129 
I have the same problem in v.3.2.15

**Comment by erhanfirat** [comment]
Same problem for me also v3.2.15

**Comment by tbredin** [comment]
Same problem here

**Comment by HeatherIris** [comment]
Same problem  for me also v3.2.15

**Comment by squarewave17** [comment]
Same here

**Comment by harshitanarayan** [comment]
I'm running into the same issue on v3.2.15

**Comment by ashalmawia** [comment]
I get the same problem, which seems to be coming from the fact that `vue-slider-component` does not support Vue3. It has dependency on `vue-class-component` [for which the issue to support Vue3 is still open](https://github.com/vuejs/vue-class-component/issues/406#issuecomment-1082817638).

You might want to explicitly mention this in the project description. This project having tag `vue3` is rather misleading.

```
└─┬ vue-slider-component@3.2.15
  └─┬ vue-property-decorator@8.5.1
    └── vue-class-component@7.2.6
```

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: vue-class-component@7.2.6
npm WARN Found: vue@3.2.37
npm WARN node_modules/vue
npm WARN   vue@"^3.2.37" from the root project
npm WARN   4 more (@vue/server-renderer, vue-property-decorator, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer vue@"^2.0.0" from vue-class-component@7.2.6
npm WARN node_modules/vue-property-decorator/node_modules/vue-class-component
npm WARN   vue-class-component@"^7.1.0" from vue-property-decorator@8.5.1
npm WARN   node_modules/vue-property-decorator
npm WARN 
npm WARN Conflicting peer dependency: vue@2.6.14
npm WARN node_modules/vue
npm WARN   peer vue@"^2.0.0" from vue-class-component@7.2.6
npm WARN   node_modules/vue-property-decorator/node_modules/vue-class-component
npm WARN     vue-class-component@"^7.1.0" from vue-property-decorator@8.5.1
npm WARN     node_modules/vue-property-decorator
```

**Comment by NightCatSama** [comment]
> Also having the same issue while trying to use this with Vue 3.2.4


@BVBAccelm vue3.x should use the `vue-slider-component@next` version.

<https://github.com/NightCatSama/vue-slider-component#-vue3x>

**Comment by aman-ghanghor** [comment]
Having same problem.

**Comment by chaegumi** [comment]
Having same problem in v3.2.24
![image](https://user-images.githubusercontent.com/9654650/211270758-89f6928c-06af-4607-9cbe-78e24ef1a391.png)

grapesjs
```javascript
    // slider
    editor.TraitManager.addType('slider', {
        createInput({ trait }) {
            const vueInst = new Vue({ render: h => h(VueSlider) }).$mount();
            const sliderInst = vueInst.$children[0];
            sliderInst.$on('change', ev => this.onChange(ev));
            this.sliderInst = sliderInst;
            return vueInst.$el;
        },
        onEvent({ component }) {
            const value = this.sliderInst.getValue() || 0;
            component.addAttributes({ value });
        },
        onUpdate({ component }) {
            const value = component.getAttributes().value || 0;
            this.sliderInst.setValue(value);
        }
    });
```

**Comment by Ouldev** [comment]
Same issue here with nuxt3 3.1.0 and vue-slider-component version 3.2.24

**Comment by vorkakashek** [comment]
Same with version 3.2.24

**Comment by ColtraneNadler** [comment]
Same issue. Unusable

**Comment by vfa-tanna** [comment]
Same with version 3.2.24

**Comment by boligolov** [comment]
Same with  version 3.2.24

**Comment by SimeonDominiq** [comment]
It worked after following this installation instruction:

`npm install vue-slider-component@next --save.`

It installed v4.1.0-beta.7. As on the read me, it should be used with caution.

**Comment by deepareddy-hub** [comment]
vue-slider-component doesnot work for the vue 3 version it is incompatible with vue 3. 
try to use this installation : npm install vue3-slider
which will resolve the issue

**Comment by kalnode** [comment]
@deepareddy-hub So to "resolve the issue" you're suggesting using a different library?

**Comment by deepareddy-hub** [comment]
> @deepareddy-hub So to "resolve the issue" you're suggesting using a different library?

Yes, because vue-slider-component facing compatibility issues for vue 3 version, so alternative library could help resolve the problem in this case.

**Comment by cyrusDev1** [comment]
I ran into some issues using vue-slider-component with Vue 3, like  "Super expression must either be null or a function".

After some testing, I found a great alternative that works smoothly with Vue 3: [@vueform/slider](https://github.com/vueform/slider).

Installation
npm install @vueform/slider
Usage with script setup
 
```
<template>
  <Slider v-model="range" :min="45" :max="2100" :tooltip="'always'" />
</template>
```

```
<script setup>
import '@vueform/slider/themes/default.css';
import Slider from '@vueform/slider';

const range = ref([45, 2100]);
</script>
```
If you're struggling with vue-slider-component on Vue 3, give this a try—it might save you some time! 😉
